src/ripple/app/hook/impl/applyHook.cpp

```diff
@@ -5511,33 +8254,96 @@ DEFINE_HOOK_FUNCTION(int64_t, float_invert, int64_t float1)
     bool const fixV3 = view.rules().enabled(fixFloatDivide);
     return float_divide_internal(float_one_internal, float1, fixV3);
 
-    HOOK_TEARDOWN();
+    WASM_HOOK_TEARDOWN();
+}
+
+DEFINE_JS_FUNCTION(JSValue, float_invert, JSValue raw_f1)
+{
+    JS_HOOK_SETUP();
+
+    auto f1 = FromJSInt(ctx, raw_f1);
+    if (!f1.has_value())
+        returnJS(INVALID_ARGUMENT);
+
+    if (*f1 == 0)
+        returnJS(DIVISION_BY_ZERO);
+    if (*f1 == float_one_internal)
+        returnJSXFL(float_one_internal);
+
+    bool const fixV3 = view.rules().enabled(fixFloatDivide);
+    int64_t const out = float_divide_internal(float_one_internal, *f1, fixV3);
+    if (out < 0)
+        returnJS(out);
+
+    returnJSXFL(out);
+
+    JS_HOOK_TEARDOWN();
 }
 
-DEFINE_HOOK_FUNCTION(int64_t, float_mantissa, int64_t float1)
+DEFINE_WASM_FUNCTION(int64_t, float_mantissa, int64_t float1)
 {
-    HOOK_SETUP();  // populates memory_ctx, memory, memory_length, applyCtx,
-                   // hookCtx on current stack
+    WASM_HOOK_SETUP();  // populates memory_ctx, memory, memory_length,
+                        // applyCtx, hookCtx on current stack
 
     RETURN_IF_INVALID_FLOAT(float1);
     if (float1 == 0)
         return 0;
     return get_mantissa(float1);
 
-    HOOK_TEARDOWN();
+    WASM_HOOK_TEARDOWN();
+}
+
+DEFINE_JS_FUNCTION(JSValue, float_mantissa, JSValue raw_f1)
+{
+    JS_HOOK_SETUP();
+
+    auto f1 = FromJSInt(ctx, raw_f1);
+    if (!f1.has_value())
+        returnJS(INVALID_ARGUMENT);
+
+    RETURNJS_IF_INVALID_FLOAT(*f1);
+
+    if (*f1 == 0)
+        returnJS(0);
```

Comment:
The success value should be returned by XFL(bigint).

```suggestion
        returnJSXFL(0);
```

Reply:
https://github.com/Xahau/xahaud/pull/481